### ISSUE TITLE
set bash variables from yaml config file

### ISSUE THREAD

**Issue by splaisan** [mikefarah/yq#529]
I would like to use a yaml file to set a number of **bash** variables and export them for my scripts to use them. 

Could you please send me in the right direction?

 Thanks in advance

The following example shows what I aim at

```
## global definitions

database:
  path: "/some/path/databases"
  name: "database.sqlite"
  dump: "structure_dump.sql"

mount:
  point: "/mnt/nas"
  path: "my/path/to/data"
```

running a bash command (?) will set and export:

```
database.path="/some/path/databases"
database.name="database.sqlite"
database.dump="structure_dump.sql"
mount.point="/mnt/nas"
mount.path="my/path/to/data"

```

**Comment by mikefarah** [comment]
This worked for me:

TLDR:
```
env $(./yq r examples/sample.yaml '**' -ppv | sed 's/: /=/' | xargs )
```

```yq r examples/sample.yaml '**' -ppv``` reads the yaml file and prints the paths and values (with a colon)

```sed 's/: /=/'``` replaces the ```: ``` with an equals

```xargs``` puts it all one one line

and finally

```env`` loads it up as env variables

**Comment by splaisan** [comment]
beautiful one-liner, but only because you explained it so well.
thank you mucho

BTW, how can I change the current '.' to '_ 'in the joining of the elements

<bash is beautiful ! who needs python?>

**Comment by mikefarah** [comment]
You mean for the variable names? Again I'd just pipe it through sed, like how the colon is replaced.

**Comment by rbreaves** [comment]
What would this look line now in v4? I am also curious about how to filter out any variables that are empty.

env $(./yq r examples/sample.yaml '**' -ppv | sed 's/: /=/' | xargs )

**Comment by ollydixon** [comment]
`version=env $(yq r pubspec.yaml '**' -ppv | sed 's/: /=/' | xargs )`

Have the following but it doesn't work, do we really need all this just to set a var? Isn't there just `yq ".property" -setVar?` for example?

**Comment by Tony-Sol** [comment]
```
yq -o p config.yaml | awk '{gsub(/\./,"_",$1)}{print toupper($1)$2$3}';
```

suits well, but if you need only e.g. `database` key, its only works for me with `grep` between `yq` and `awk` - there is no `-ppv` option anymore :c

**Comment by cioban** [comment]
I've accomplished this by using the same syntax that I used on jq:
`eval $(yq -r 'to_entries | .[] | .key + "=" + (.value | @sh)' file.yaml`
